eclipse-theia/theia
# [task] fix circular dependencies from 'task' extension

#### What it does


Fixes the circular dependencies present from the `@theia/task` extension:
- Remove the circular dependency.
- Export the moved classes from the original file not to break consumers. 


| Before | After |
|:---|:---|
|![image](https://user-images.githubusercontent.com/40359487/70907526-193bca80-1fd7-11ea-88fe-498293278b7d.png)|![image](https://user-images.githubusercontent.com/40359487/70907470-ef82a380-1fd6-11ea-8a72-8012d37b1ec5.png)|

#### How to test


- Verify that CI successfully passes (build & tests of the `@theia/task` extension).
- Verify that clicking the `Configure` icon (cog) still works correctly when selecting `Run Tasks...`.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>
